opencodelists/forms.py

```diff
@@ -42,9 +43,23 @@ def clean_new_password2(self):
         return password2
 
 
+def form_field_from_model(model_class, field_name, **kwargs):
+    """Return a forms.Field instance derived from a models.Field,
+    including the validators attached to the model field.
+
+    Note:
+    - `formfield` does not automatically inherit validators, as they may
+      not always apply to a form field.
+    - It is the caller's responsibility to ensure the validators are
+      appropriate for their specific form."""
+    model_field = model_class._meta.get_field(field_name)
+    field_class = model_field.formfield
+    return field_class(validators=model_field.validators, **kwargs)
+
+
 class CodelistCreateForm(forms.Form):
     owner = forms.ChoiceField()
-    name = forms.CharField(max_length=255, label="Codelist name")
+    name = form_field_from_model(Handle, "name", label="Codelist name")
```

Comment:
The original issue (#2576) mentioned frontend validation; maybe something like this would suffice?

```suggestion
    name = form_field_from_model(
        Handle,
        "name",
        label="Codelist name",
        widget=forms.TextInput(
            attrs={
                "pattern": ".*[A-Za-z0-9].*",
                "title": "The name of the codelist (must include at least one or more alphanumeric characters)",
            }
        ),
    )
```

The `.*` in `pattern` is required because `pattern` matches the entire string, and we only require that there are alphanumeric characters in the string.